### ISSUE TITLE
Requests: Sorting Task list and Search box

### ISSUE THREAD

**Issue by Alphos2000** [kapitainsky/RcloneBrowser#21]
I just found your fork and god, that task list is was I was really missing! Thank you!

I would like to request two things if possible:

1 - Any way of sort the Task list. They get sorted by the date they are created. Maybe some drag and drop or arrow buttons to change order would be nice, specially when you have a lot and want to "group" them together.

2 - Could be possible to add any kind of search box / filtering box so, when you load the file list of one or more folders, the list get filtered showing only the files containing the text in their filename?

Thank you!

**Comment by kapitainsky** [comment]
Thanks for feedback. I will add this to wish list. Atm I am trying to fix windows mount issue as main priority but I will look into this in the future.

**Comment by kapitainsky** [comment]
In terms of search box - if your goal is to transfer only specific files you can do it now by providing extra options for your transfer, e.g.

`--include *.{png,jpg}` to include all png and jpg files and no others.

check this site for details https://rclone.org/filtering/

**Comment by kapitainsky** [comment]
> 1 - Any way of sort the Task list. They get sorted by the date they are created. Maybe some drag and drop or arrow buttons to change order would be nice, specially when you have a lot and want to "group" them together.

This should definitely have priority

**Comment by Alphos2000** [comment]
First of all, thank you so much for the improvements. 

Maybe I didn't express very well about the search box. Sorry, english isn't my maternal language. About the search box, what I was thinking about is a search box so, when I expand in my google drive my folder "movies" with 6000 files, I could write on it "Lord of" so it will filter my 6000 movies list and show only the files containing "lord of" in the file name.

**Comment by kapitainsky** [comment]
What works is that you can actually type and app will take you to the right file/folder...

So in your case you just start typing "Lord" and you will get to the right place

**Comment by dia3olik** [comment]
also using a locally cached db it should be possible to do a real time search without impacting the cloud servers at all ;-)

**Comment by kapitainsky** [comment]
yeap. PR welcomed :)

**Comment by redfanglicker** [comment]
Is the search function implemented?

**Comment by kapitainsky** [comment]
No. There is no search function.  Rclone does not have functionality like this.

**Comment by redfanglicker** [comment]
> No. There is no search function. Rclone does not have functionality like this.
> […](#)
> On 4 Jan 2020, at 11:31, redfanglicker ***@***.***> wrote: ﻿ Is the search function implemented? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

So is there any way to search for file with in a google drive?

**Comment by kapitainsky** [comment]
Not in rclone browser

**Comment by kapitainsky** [comment]
You could mount google drive locally and use your OS file search

**Comment by kapitainsky** [comment]
In general issue it that cloud storage services do not provide “search” API. Sometimes you can find it in their web based versions - like google drive. But often it is like ftp - you can list content of specific folder. That’s it. To have search from within rclone browser would require traversing all directories - some people like me have millions of files in “the cloud”. It would be very slow. Definitely it is doable if somebody is ready for the challenge:)

**Comment by kapitainsky** [comment]
In the next release tasks can be sorted. Search is not on the map - it is not something rclone supports and I dont see sensible way to implement it in Rclone Browser

**Comment by LinuxOnTheDesktop** [comment]
> Search is not on the map - it is not something rclone supports

Well, rclone does do what it calls ['filtering'](
https://rclone.org/filtering/).
